plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java

```diff
@@ -0,0 +1,597 @@
+                .addRoundTrip(inputType, inputLiteralFactory.apply(epoch, fixedOffsetWest), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(epoch, fixedOffsetWest))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(afterEpoch, UTC), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(afterEpoch, UTC))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(afterEpoch, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(afterEpoch, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(afterEpoch, fixedOffsetEast), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(afterEpoch, fixedOffsetEast))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(afterEpoch, fixedOffsetWest), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(afterEpoch, fixedOffsetWest))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInJvmZone, UTC), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInJvmZone, UTC))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInJvmZone, jvmZone), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInJvmZone, jvmZone))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInJvmZone, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInJvmZone, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInVilnius, UTC), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInVilnius, UTC))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInVilnius, vilnius), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInVilnius, vilnius))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeDoubledInVilnius, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeDoubledInVilnius, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInJvmZone1, UTC), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInJvmZone1, UTC))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInJvmZone1, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInJvmZone1, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInJvmZone2, UTC), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInJvmZone2, UTC))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInJvmZone2, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInJvmZone2, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInVilnius, kathmandu), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInVilnius, kathmandu))
+                .addRoundTrip(inputType, inputLiteralFactory.apply(timeGapInKathmandu, vilnius), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(timeGapInKathmandu, vilnius));
```

Comment:
Could you try avoiding redundant code? e.g.
```java
        SqlDataTypeTest tests = SqlDataTypeTest.create();
        tests.addRoundTrip(inputType, "NULL", TIMESTAMP_TZ_MILLIS, "CAST(NULL AS TIMESTAMP WITH TIME ZONE)");
        for (ZoneId zoneId : ImmutableList.of(UTC, kathmandu, fixedOffsetEast, fixedOffsetWest)) {
            tests.addRoundTrip(inputType, inputLiteralFactory.apply(beforeJulianGregorianSwitch, zoneId), TIMESTAMP_TZ_MILLIS, expectedLiteralFactory.apply(beforeJulianGregorianSwitch, zoneId));
...
        }
        return tests;
```

Reply:
Good suggestion.

Reply:
@tangjiangling Is there any reason you didn't move some roundtrips?

Reply:
I want the test cases to be consistent with other connectors (e.g. PostgreSQL, Oracle).

Reply:
Let's move these cases to loop. Consistency within a method should win.

Reply:
done